src/krfctl/freebsd/freebsd.c

```diff
@@ -0,0 +1,99 @@
+#include <stdio.h>
+#include <string.h>
+#include <sys/errno.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <err.h>
+#include <string.h>
+#include <sys/sysctl.h>
+
+#include "../krfctl.h"
+
+/* control will interpret any number larger than its syscall table
+ * as a command to clear all current masks.
+ * it's a good bet that linux will never have 65535 syscalls.
+ */
+#define CLEAR_MAGIC 65535
+
+#define CONTROL_NAME "krf.control"
+#define RNG_STATE_NAME "krf.rng_state"
+#define PROBABILITY_NAME "krf.probability"
+#define LOG_FAULTS_NAME "krf.log_faults"
+#define TARGETING_NAME "krf.targeting"
+
+void fault_syscall(const char *sys_name) {
+  const char *sys_num;
+  unsigned int syscall;
+
+  if (!(sys_num = lookup_syscall_number(sys_name))) {
+    errx(1, "couldn't find syscall %s", sys_name);
+  }
+
+  if (sscanf(sys_num, "%u", &syscall) != 1) {
+    err(errno, "weird syscall number");
+  }
+
+  if (sysctlbyname(CONTROL_NAME, NULL, NULL, &syscall, sizeof(syscall)) < 0) {
+    errx(errno, "faulting for %s unimplemented", sys_name);
```

Comment:
Nit: Test for `errno == EOPNOTSUPP` here and specialize the error message, like we do for Linux.